api/mapping/services_rules.py

```diff
@@ -393,7 +393,180 @@ def get_rules_by_scan_report_concept(scan_report_concept_id):
+        if obj.content_type.model_class() is ScanReportValue
+    ]
+    #make a batch call to the ORM again..
+    scan_report_values = {
+        obj.id:obj.value
+        for obj in list(ScanReportValue.objects.filter(pk__in=scan_report_values))
+    }
+
+    #get all destination field ids
+    destination_fields = [
+        obj.omop_field_id
+        for obj in structural_mapping_rules
+    ]
+    #batch call
+    destination_fields = {
+        obj.id:obj
+        for obj in list(OmopField.objects.filter(pk__in=destination_fields))
+    }
+
+    #again with destination table
+    destination_tables = [
+        obj.table_id for obj in destination_fields.values()
+    ]
+    destination_tables = {
+        obj.id:obj for obj in list(OmopTable.objects.filter(pk__in=destination_tables))
+    }
+
+    #and sources....
+    source_fields = [
+        obj.source_field_id for obj in structural_mapping_rules
+    ]
+    source_fields = {
+        obj.id:obj for obj in list(ScanReportField.objects.filter(pk__in=source_fields))
+    }
+    source_tables = [
+        obj.scan_report_table_id for obj in source_fields.values()
+    ]
+    source_tables = {
+        obj.id:obj for obj in list(ScanReportTable.objects.filter(pk__in=source_tables))
+    }
+
+    #now looop over the rules to actually create the list version of the rules
+    rules = []
+    for rule in structural_mapping_rules:
+
+        #get the fields/tables from the loop up lists
+        #the speed up comes from here as we dont need to keep hitting the DB to get this data
+        #we've already cached it in these dictionaries by making a batch call
+        destination_field = destination_fields[rule.omop_field_id]
+        destination_table = destination_tables[destination_field.table_id]
+
+        source_field = source_fields[rule.source_field_id]
+        source_table = source_tables[source_field.scan_report_table_id]
+
+        #get the concepts again
+        scan_report_concept_id = rule.concept_id
+        scan_report_concept = scan_report_concepts[rule.concept_id]
+        concept_id = scan_report_concept.concept_id
+
+        #work out if we need term_mapping or not
+        term_mapping = None
+        if 'concept_id' in destination_field.field:
+            if scan_report_concept.content_type.model_class() is ScanReportValue:
+                term_mapping = {scan_report_values[scan_report_concept.object_id]:concept_id}
+            else:
+                term_mapping = concept_id
+            
+        rules.append(
+            {
+                'rule_id':scan_report_concept_id,
+                'destination_table':destination_table,
+                'destination_field':destination_field,
+                'source_table':source_table,
+                'source_field':source_field,
+                'term_mapping':term_mapping
+            })
+            
+    return rules
+
+def get_mapping_rules_json_batch(structural_mapping_rules):
```

Comment:
Please comment with how this is different to `_list`.

Reply:
yeh good point, names are stupid. In the last commit I added more detail to the docstring and then called this function `get_mapping_rules_json` and removed the old `get_mapping_rules_json`

Summary:
* `get_mapping_rules_list`  --> gets a list of mapping rules that can be displayed by the template html 
* `get_mapping_rules_json`  --> first gets `get_mapping_rules_list`   and then converts it into a dictionary/json in the format needed by the ETLTool